pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -1525,6 +1530,7 @@ public int messagesCount() {
     @Override
     public void connectionOpened(final ClientCnx cnx) {
         previousExceptions.clear();
+        chunkMaxMessageSize = Math.min(chunkMaxMessageSize, ClientCnx.getMaxMessageSize());
```

Comment:
If we assume that `ClientCnx.getMaxMessageSize()` may change on new connections. Consider the case :
1. Previous connection: `conf.getChunkMaxMessageSize()` == 100 and `ClientCnx.getMaxMessageSize() == 50`, `chunkMaxMessageSize` init as 50;
2. New connection: `chunkMaxMessageSize` == 50 and `ClientCnx.getMaxMessageSize()` == 200, the result is 50, but it should be 100.

Reply:
> max-size is not something broker specific but it's bk specific

We may have different settings on max-size both in broker and bookie, as long as broker limit < bookie limit. 
And we can update broker level setting by changing the config file and restart. So it should be handled.